### ISSUE TITLE
Add rate limit for unsolicited/duplicate replies or mentions

### ISSUE THREAD

**Issue by lucaswerkmeister** [IBBoard/cawbird#409]
**Is your feature request related to a problem? Please describe.**
If a user uses Cawbird to send tweets with unsolicited or duplicate replies or mentions in rapid succession, Twitter may respond not by blocking the user, but by blocking the application used to send the tweets. This happened to the Cawbird for Arch Linux application with as few as **58 mentions in 14 tweets in 4 minutes**, and took a week to resolve with support.

**Describe the solution you'd like**
I believe Cawbird has to apply a strict rate limit for such tweets, otherwise a single malicious user can effectively break Cawbird for all other users using the same application (e.g. from the same Linux distribution).

I suggest a rate limit of 10 mentions per 5 minutes.

**Describe alternatives you've considered**
It would of course be nice if Twitter was more considerate in its blocks, but we can’t control that, and my interactions with Twitter Support so far do not inspire any trust in that course of action in me. They are unwilling to even acknowledge that a Cawbird application (in my case, Cawbird for Arch Linux) may be used by several users.

**Comment by IBBoard** [comment]
I think we'd have to do "10 *new* mentions in 5 minutes" (or some other metric) so that we let people reply to threads with lots of pre-existing people in. The examples that you caught were all "spamming random people"

I was thinking about some kind of "similarity index" as well. How often do people legitimately post similar replies? If we can do a "bucket of words" for the last few tweets then we can catch people posting "predominantly identical" content. If we say "90% similar text (excluding handles)" and turn that into a word count and do `floor()` (so a four word tweet triggers then three words match in the next tweet) then it *might* catch some stuff without being annoying. Maybe we do a non-linear back-off for those where we allow a high overlap once then say "wait 5 seconds" then "wait 25 seconds" etc.

But then could someone hit the [Thread Reader App](https://twitter.com/threadreaderapp) repeatedly in a short period? Maybe not because they're unlikely to see lots of threads that quickly? Or are there any "Play By Tweet" games etc where you might want to do the same action repeatedly and we'd incorrectly flag that on "similarity"? (But then maybe similarity is only a problem when messaging *new* people?)

(What's particularly annoying here - other than Twitter Support's complete inability to understand that we're a third-party client and have no control over what people post! - is that I've just translated what we think are the "offending" tweets and it appears to be someone asking politicians whether they're listening to a petition! So it's not even abusive spam, it's potentially just an activist 🙄)

Looping @CodedOre in as well, because it's going to be a problem for v2.0 to work with as well.

**Comment by lucaswerkmeister** [comment]
> I think we'd have to do "10 _new_ mentions in 5 minutes" (or some other metric) so that we let people reply to threads with lots of pre-existing people in. The examples that you caught were all "spamming random people"

Yeah, my interpretation of “unsolicited mention” is that it’s about accounts who aren’t in a conversation/thread yet. Replying to someone hopefully isn’t considered spammy.
